### PR TITLE
Automate quick start steps in setup.sh and teardown.sh

### PR DESCRIPTION
**Pull request by JulianWei**

### CODE REVIEW

File: deployments/quick-start/setup.sh [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+vault write root-ca/config/urls issuing_certificates="http://vault:8200/v1/root-ca/ca" crl_distribution_points="http://vault:8200/v1/root-ca/crl"
+
+# 2.3 Create the Intermediate Certificate Authority
+# Create the Intermediate CA that expires in 5 years: 
+vault mount -path=intermediate-ca -max-lease-ttl=43800h pki
+
+# Generate a Certificate Signing Request: 
+http POST http://127.0.0.1:8200/v1/intermediate-ca/intermediate/generate/internal X-Vault-Token:vault-root-token common_name="Intermediate CA" ttl=43800h exclude_cn_from_sans=true | jq -r .data.csr > intermediate.csr
+
+
```
Comment (F21) [inline deployments/quick-start/setup.sh]: Only 1 new line here.

File: deployments/quick-start/setup.sh [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+# Send the stored certificate back to Vault: 
+vault write intermediate-ca/intermediate/set-signed certificate=@signed.crt
+
+#Set up URLs: 
+vault write intermediate-ca/config/urls issuing_certificates="http://vault:8200/v1/intermediate-ca/ca" crl_distribution_points="http://vault:8200/v1/intermediate-ca/crl"
+
+#Create a role to allow Kubernetes-Vault to generate certificates: 
+vault write intermediate-ca/roles/kubernetes-vault allow_any_name=true max_ttl="24h"
+
+
```
Comment (F21) [inline deployments/quick-start/setup.sh]: Only 1 new line here.

File: deployments/quick-start/setup.sh [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+#4. Deploy a sample app
+
+#4.1 Prepare the manifest and deploy
+
+#Inspect deployments/quick-start/sample-app.yaml and update the role id in the deployment.
+kubectl apply -f sample-app.yaml
+
+#5. Confirm that each pod of the sample app received a Vault token
+echo View the logs using the Kubernetes dashboard or kubectl logs mypod and confirm that each pod receive a token. The token and various other information related to the token should be logged.
+
```
Comment (F21) [inline deployments/quick-start/setup.sh]: Remove the extra new lines.

File: deployments/quick-start/kubernetes-vault.yaml [hunk 4]
```diff
@@ -60,4 +60,4 @@ spec:
       volumes:
         - name: config-volume
           configMap:
-            name: kubernetes-vault
\ No newline at end of file
+            name: kubernetes-vault
```
Comment (F21) [inline deployments/quick-start/kubernetes-vault.yaml]: Please rebase against master as the changes to this file is not necessary.

**Review by F21 — CHANGES_REQUESTED** [review]
Looks great! Can you please squash your commits? I'll try it out later today or tomorrow and merge if it's all good 😄

File: deployments/quick-start/kubernetes-vault.yaml (hunk 4 above)
Comment (F21) [reply deployments/quick-start/kubernetes-vault.yaml]: Can you exclude the change to this file?

**Comment by JulianWei** [comment]
I updated my pull request, please take a look.

**Comment by F21** [comment]
Thanks! 👍 I'll spin up a minikube cluster today or tomorrow to test it out.

**Comment by F21** [comment]
@JulianWei I just tested out the setup and teardown scripts and they work great! 

However, I am seeing this error (it does not negatively affect the setup, but it would still be good to fix).

See the complete log:
```
$ ./setup.sh
service "vault" created
deployment "vault" created
Vault pod name: vault-3485173613-05575
wait 1 second for Vault up and running
./setup.sh: 39: ./setup.sh: maxWaitSecond--: not found <---------------------
Vault pod status Running
nohup: appending output to 'nohup.out'
Successfully authenticated! You are now logged in.
token: vault-root-token
token_duration: 0
token_policies: [root]
Successfully mounted 'pki' at 'root-ca'!
Key             Value
---             -----
certificate     -----BEGIN CERTIFICATE-----
MIIC9DCCAdygAwIBAgIUW0MogXyKdy0r29DKboKCCg0eeg0wDQYJKoZIhvcNAQEL
BQAwEjEQMA4GA1UEAxMHUm9vdCBDQTAeFw0xNzA1MDYwNzQwMjRaFw0yNzA1MDQw
NzQwNTRaMBIxEDAOBgNVBAMTB1Jvb3QgQ0EwggEiMA0GCSqGSIb3DQEBAQUAA4IB
DwAwggEKAoIBAQDz8suif4i/oNsP7p2HGyukW3IosdRDeUg980OcD6lSBJsLrBCB
8U70voxcUyX6JOrXXLFirEs5tjS2VJ9KPB0vQCUERqmPHJBP/Wa33YU9KhLA7yJt
tFYVEVpXa+UCWcWoB917Qgp2iLK6FTs0PfqHfjHTisVcVPw4Ueil7IvdIZdsweOg
0EHQutW5VSKPwipSHTzShDhoW6DoRO6J4webXt/Z2Eg+feKnLk3lOvwVOicAbtwj
a9IKsmLpe4PR0lSltXnFVAwvRicopQLy95bIOxMDHrfgFxOlJZjAt/Sg1ASdhnmV
fhfBbNumIyxBEmJiKGgIJ0xI2iMocYIzYWwzAgMBAAGjQjBAMA4GA1UdDwEB/wQE
AwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBRhCgLqs0hmJphE4Ptg94OQ
np2aJzANBgkqhkiG9w0BAQsFAAOCAQEA77zCYv2jhS9rWsWJcFjIitHN6SsLqIeU
c8yC3YG7ThKiWgc/rrxiL76YVYCfQKOi2xwqFqHryFnWK66rGksnyPo32lYB5Eaq
9tckju2LZCtqSHnhJ4lgK/iyyYJg+CqfPqT2LMv9QYWcMhYHdT54mwbShRQdR5u4
vUFcq1ZQrDfLBdXwCil46+Q9liycsUEJbT8Z8ya7uHXGAuBPdec8gscjIZaGwjH+
RnEa4zZSkf2ZHRLzWeiP+meGVm0Ks7gCaFTtvFQiC2/DIJMlr6hndnYIH2UzpsYw
MjwC8KcaC7KnZi0UqZ76hX4Sx26TtCc5hA5gWdMIrSE5ZxNORPE0qg==
-----END CERTIFICATE-----
expiration      1809416454
issuing_ca      -----BEGIN CERTIFICATE-----
MIIC9DCCAdygAwIBAgIUW0MogXyKdy0r29DKboKCCg0eeg0wDQYJKoZIhvcNAQEL
BQAwEjEQMA4GA1UEAxMHUm9vdCBDQTAeFw0xNzA1MDYwNzQwMjRaFw0yNzA1MDQw
NzQwNTRaMBIxEDAOBgNVBAMTB1Jvb3QgQ0EwggEiMA0GCSqGSIb3DQEBAQUAA4IB
DwAwggEKAoIBAQDz8suif4i/oNsP7p2HGyukW3IosdRDeUg980OcD6lSBJsLrBCB
8U70voxcUyX6JOrXXLFirEs5tjS2VJ9KPB0vQCUERqmPHJBP/Wa33YU9KhLA7yJt
tFYVEVpXa+UCWcWoB917Qgp2iLK6FTs0PfqHfjHTisVcVPw4Ueil7IvdIZdsweOg
0EHQutW5VSKPwipSHTzShDhoW6DoRO6J4webXt/Z2Eg+feKnLk3lOvwVOicAbtwj
a9IKsmLpe4PR0lSltXnFVAwvRicopQLy95bIOxMDHrfgFxOlJZjAt/Sg1ASdhnmV
fhfBbNumIyxBEmJiKGgIJ0xI2iMocYIzYWwzAgMBAAGjQjBAMA4GA1UdDwEB/wQE
AwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBRhCgLqs0hmJphE4Ptg94OQ
np2aJzANBgkqhkiG9w0BAQsFAAOCAQEA77zCYv2jhS9rWsWJcFjIitHN6SsLqIeU
c8yC3YG7ThKiWgc/rrxiL76YVYCfQKOi2xwqFqHryFnWK66rGksnyPo32lYB5Eaq
9tckju2LZCtqSHnhJ4lgK/iyyYJg+CqfPqT2LMv9QYWcMhYHdT54mwbShRQdR5u4
vUFcq1ZQrDfLBdXwCil46+Q9liycsUEJbT8Z8ya7uHXGAuBPdec8gscjIZaGwjH+
RnEa4zZSkf2ZHRLzWeiP+meGVm0Ks7gCaFTtvFQiC2/DIJMlr6hndnYIH2UzpsYw
MjwC8KcaC7KnZi0UqZ76hX4Sx26TtCc5hA5gWdMIrSE5ZxNORPE0qg==
-----END CERTIFICATE-----
serial_number   5b:43:28:81:7c:8a:77:2d:2b:db:d0:ca:6e:82:82:0a:0d:1e:7a:0d

Success! Data written to: root-ca/config/urls
Successfully mounted 'pki' at 'intermediate-ca'!
Success! Data written to: intermediate-ca/intermediate/set-signed
Success! Data written to: intermediate-ca/config/urls
Success! Data written to: intermediate-ca/roles/kubernetes-vault
Successfully enabled 'approle' at 'approle'!
Success! Data written to: auth/approle/role/sample-app
Policy 'kubernetes-vault' written.
Success! Data written to: auth/token/roles/kubernetes-vault
service "kubernetes-vault" created
configmap "kubernetes-vault" created
deployment "kubernetes-vault" created
deployment "sample-app" created
View the logs using the Kubernetes dashboard or kubectl logs mypod and confirm that each pod receive a token. The token and various other information related to the token should be logged.
```

Another concern of mine is that we are only waiting for 5 seconds for the Vault image to download and deploy.

Since the Vault image is around 67MB, I suggest what we increase the timeout for download/deploy to 10 minutes, so that it can have enough time to download over slow connections.

**Pull request closed without merging**